### ISSUE TITLE
[docs] - Support for RSS

### ISSUE THREAD

**Issue by ericdallo** [emacs-lsp/lsp-mode#2184]

**Comment by ericdallo** [comment]
I'm waiting for this PR to add to lsp-mode: https://github.com/squidfunk/mkdocs-material/pull/1928

**Comment by yyoncho** [comment]
I think we are done here but I am unable to find RSS on the website?

**Comment by ericdallo** [comment]
Yeah, me too, maybe @Guts knows why :thinking: ?

**Comment by Guts** [comment]
As [documentation](https://guts.github.io/mkdocs-rss-plugin/) says (but maybe it's not clear enough, I'll reformulate):

>    feed_rss_created.xml for latest created pages
>   feed_rss_updated.xml for latest updated pages

So, in your case, RSS files can be viewed:

- Into the gh-pages branch (to check for example):

  - [created content](https://github.com/emacs-lsp/lsp-mode/blob/gh-pages/feed_rss_created.xml)
  - [updated content](https://github.com/emacs-lsp/lsp-mode/blob/gh-pages/feed_rss_updated.xml)

- served on your website:

  - [created content](https://emacs-lsp.github.io/lsp-mode/feed_rss_created.xml)
  - [updated content](https://emacs-lsp.github.io/lsp-mode/feed_rss_updated.xml)

**Comment by ericdallo** [comment]
Thanks @Guts, I think @yyoncho was referring to the RSS icon, if it should appear somewhere in the webpage

**Comment by yyoncho** [comment]
> Thanks @Guts, I think @yyoncho was referring to the RSS icon, if it should appear somewhere in the webpage

Maybe we should link it from the footer, no?

**Comment by Guts** [comment]
> Thanks @Guts, I think @yyoncho was referring to the RSS icon, if it should appear somewhere in the webpage

The plugin doesn't insert anything into the HTML. You are free to add an icon in the footer (example here https://github.com/geotribu/website/blob/master/mkdocs.yml#L81-L83) or a Feedly button or any image with a link to the feed 😃.

**Comment by clemera** [comment]
It would be good to reference the feed in the header so browsers and other tools can detect the feed, something like:

```html
<link rel="alternate" type="application/rss+xml" title="RSS" href="rss.xml">
```

**Comment by clemera** [comment]
It seems the feed is only for the pages and not the blog?

**Comment by ericdallo** [comment]
Yeah, for some reason it didn't include the blog/tutorials, do you know why @Guts ?

**Comment by Guts** [comment]
I do not know your docs structure so hard to help, but looking at your CI build, it seems to be a lack of git history during build: https://github.com/emacs-lsp/lsp-mode/runs/1136396394#step:8:126 (the git revision date uses the same git logic than the rss plugin).

These plugins use the git log to determine creation and update dates. if there is no git log during the docs build, it fallbacks to the build date, so every page will have the same dates... It's easy to see looking at the pages footer or `pubDate` tag in the RSS output.

See: https://github.com/timvink/mkdocs-git-revision-date-localized-plugin#note-when-using-build-environments

Why using Docker builds to deploy static files? You could use a simple CI workflow using directly a requirements file for your plugins and using the `mkdocs gh-deploy` included command. For example, for the plugin documentation: https://github.com/Guts/mkdocs-rss-plugin/blob/master/.github/workflows/docs_to_gh_pages.yml.

**Comment by ericdallo** [comment]
@Guts it's because all [laguages pages](https://emacs-lsp.github.io/lsp-mode/page/languages/) are generated dynamically every build, but the other pages are versioned.

**Comment by Guts** [comment]
Okay got it, si that explains why from the plugin point of view languages pages are the latest created and updated content.

I'm working on a feature to handle the meta.date attribute for overriding git log. It could help but I'll need some free time...

**Comment by ericdallo** [comment]
No problem, thanks @Guts

**Comment by Guts** [comment]
Hi there.

> I'm working on a feature to handle the meta.date attribute for overriding git log. It could help but I'll need some free time...

Just t let you know: feature achieved and published in the 0.10.0 plugin version.

**Comment by ericdallo** [comment]
Nice, I'll take a look later
FYI @yyoncho

**Comment by yyoncho** [comment]
@ericdallo are we done here?

**Comment by ericdallo** [comment]
I don't think so, I guess we need to add this date meta do generated pages, right @Guts ?

**Comment by Guts** [comment]
> I don't think so, I guess we need to add this date meta do generated pages, right @Guts ?

That's it. WIth your dynamic workflow, you may want to generate the created and/or the update date on the fly.

**Comment by ericdallo** [comment]
Nice, I'll take a look on this later, thanks!
